### PR TITLE
Implement remove and edit

### PR DESCRIPTION
**Pull request by hopeforkoami**
in this branch , I
-Remove all hardcoded items from the tasks array.
-Create a new JavaScript file for the new functionality.
-Implement a function for adding a new task (add a new element to the array).
-Implement a function for deleting a task (remove an element from the array).
-Implement a function for editing task descriptions.
-By default new tasks have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
-Deleting a task updates all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
-All changes to the To Do List are saved in local storage.

thanks to the reviewer

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,46 +1,122 @@
...
   }
 }
 
+function completeTask(index) {
+  hideComponent(document.querySelector(`.uncomplete${index}`));
+  displayComponent(document.querySelector(`.complete${index}`));
+  hideComponent(document.querySelector(`.details${index}`));
+  displayComponent(document.querySelector(`.delete${index}`));
+  displayLineThroughComponent(document.querySelector(`.input${index}`));
+  // console.log(tasklist.localData[index-1].getCompleted());
```
Comment (tresorsawasawa) [inline src/index.js]: - [ ] Tom make your codebase cleaner and more professional, kindly make sure your remove all the commented code lines.

File: src/modules/tasklist.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    this.localData.splice(index, 1);
+  }
+
+  localStrore() {
+    localStorage.setItem('taskList', JSON.stringify(this.localData));
+  }
+
+  clearCompletedTasks() {
+    const temps = [];
+    // console.log('inside the task list and the task'+key+1+' statut == '+task.completed);
```
Comment (tresorsawasawa) [inline src/modules/tasklist.js]: - [ ] As mentioned in the comment above, kindly make sure you remove all the commented code lines from your code base, to make it cleaner and professional.

File: src/index.js [hunk 3]
```diff
@@ -1,46 +1,122 @@
...
+}
+
+function uncompleteTask(index) {
+  hideComponent(document.querySelector(`.complete${index}`));
+  displayComponent(document.querySelector(`.uncomplete${index}`));
+  hideComponent(document.querySelector(`.delete${index}`));
+  displayComponent(document.querySelector(`.details${index}`));
+  removeLineThroughComponent(document.querySelector(`.input${index}`));
+  tasklist.localData[index - 1].setCompleted(false);
+}
```
Comment (tresorsawasawa) [inline src/index.js]: - [ ] Kindly make sure the user is able to `delete` a task without depending on the `checkbox` button status. Right now, a **user** can only delete a **task or item** if only the `checkbox` button is **checked** but I think deleting a task will not be dependent on the  `checkbox` button status. Make sure you fix that.

a_todo

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @hopeforkoami 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sumairq — APPROVED** [review]
### Hi @hopeforkoami  , 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good commit history
✔️Linters are passing
✔️Very professional README
✔️Correct Github flow
✔️Descriptive PR
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**